vmicore/src/lib/os/windows/KernelOffsets.cpp

```diff
@@ -45,7 +45,10 @@ namespace VmiCore::Windows
                                 .Right = vmiInterface->getKernelStructOffset("_RTL_BALANCED_NODE", "Right")},
             .fileObject = {.FileName = vmiInterface->getKernelStructOffset("_FILE_OBJECT", "FileName")},
             .section = {.controlArea = vmiInterface->getKernelStructOffset("_SECTION", "u1")},
-            .kprocess = {.DirectoryTableBase = vmiInterface->getKernelStructOffset("_KPROCESS", "DirectoryTableBase")},
+            .kprocess = {.kernelDirectoryTableBase =
+                             vmiInterface->getKernelStructOffset("_KPROCESS", "DirectoryTableBase"),
+                         .userDirectoryTableBase =
+                             vmiInterface->getKernelStructOffset("_KPROCESS", "UserDirectoryTableBase")},
```

Comment:
This will throw if `UserDirectoryTableBase` is not present in the offsets file. I suggest using libvmi's `vmi_get_winver` API and depending on the version to either retrieve the offset for `UserDirectoryTableBase` or for `DirectoryTableBase`.